Gabriel/cs/cf 181 add to planner transition

gtangelo opened the pull request:
Changes made:
- Refactored file structure and cleaned up course description components
- Fix bugs associated with the add to planner button (CF-174 - text placement when removing course, CF-186 - flashing when course is added)
- CF-181 - Added correct transition when adding course to planner i.e.

![image](https://user-images.githubusercontent.com/65492346/165514506-f28d22c8-8087-4be8-a477-37374d5d3252.png)
![image](https://user-images.githubusercontent.com/65492346/165514533-62cd45de-40fe-4f73-979e-ff80f6f18bd4.png)
![image](https://user-images.githubusercontent.com/65492346/165514551-25a8c37a-7de8-44d8-860c-540be82274ab.png)
![image](https://user-images.githubusercontent.com/65492346/165514589-7ac2d296-277f-430e-8a8a-12230b4828b5.png)

gtangelo committed:
removed dropdown

gtangelo committed:
Merge branch 'dev' into gabriel/cs/CF-181-add-to-planner-transition

gtangelo committed:
CF-186 CF-181 CF-174 fix bugs associated when clicking add to planner button

gtangelo committed:
Refactored files in course description

gtangelo committed:
Merge branch 'dev' into gabriel/cs/CF-181-add-to-planner-transition

Peedee2002 commented on frontend/src/pages/CourseSelector/courseDescription/AddToPlannerButton.jsx:
```diff
@@ -0,0 +1,95 @@
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (!!coursesInPlanner[id] !== addedCourseInPlanner) {
```
this is a guard, so use an early return

Peedee2002 commented on frontend/src/pages/CourseSelector/courseDescription/AddToPlannerButton.jsx:
```diff
@@ -0,0 +1,95 @@
+          setLoading(false);
+          setAddedCourseInPlanner(true);
+        }, 1000);
+      } else if (!coursesInPlanner[id]) {
```
these two clauses are the same in every respect except for setAddedCourseInPlanner - use a turnary statement to differentiate between the two

Peedee2002 commented on frontend/src/pages/CourseSelector/courseDescription/AddToPlannerButton.jsx:
```diff
@@ -0,0 +1,95 @@
+      },
+    };
+    dispatch(addToUnplanned(data));
+    setLoading(true);
```
this block is used a lot - refactor into a function.

Peedee2002 commented on frontend/src/pages/CourseSelector/courseDescription/CourseDescription.jsx:
```diff
@@ -119,81 +68,18 @@ const CourseDescription = ({ structure }) => {
-                  )}
-                  loading={loading}
-                  onClick={() => {
-                    addToPlanner("Uncategorised");
```
where did this logic go?

gtangelo replied:
Removed the dropdown for now since we aren't considering the course's type (i.e. Core Course, Gen Ed) for the term planner atm. Will revisit this down the road

Peedee2002 reviewed (changes requested):
a few stylistic changes here, but i like the change :)

gtangelo committed:
PR feedback

gtangelo commented:
This may create a conflict with your changes on the deselect so feel free to merge yours in first then I can fix any potential issues that may arise! @Peedee2002

gtangelo committed:
Merged dev and streamlined planner reducers to reuse other reducer logic, fix lag with framer motion dropdown

gtangelo committed:
Merge branch 'dev' into gabriel/cs/CF-181-add-to-planner-transition

gtangelo commented:
Updated changes:
- reuse logic of `unschedule` reducer in `unscheduleAll` in `plannerSlice.js` rather than repeating code to unschedule all courses
- moved logic of removing several courses into a reducer function. This fixes the lag present with framer motion if the course removed removes alot of courses that require it. i.e. removing 'COMP1511' causes some lag with the course selector menu dropdown when the user has already added several comp courses

Peedee2002 reviewed (approved):
looks good to me
